Lib/test/test_list.py

```diff
@@ -118,6 +118,19 @@ def test_list_resize_overflow(self):
         with self.assertRaises((MemoryError, OverflowError)):
             lst *= size
 
+    def test_repr_mutate(self):
+        class Obj:
+            @staticmethod
+            def __repr__():
+                try:
+                    mylist.pop()
```

Comment:
`__repr__()` doesn't use `self` (static method) and so doesn't crash if `self` becomes a dangling pointer (borrowed reference). If `__repr__()` uses `self (regular method), it works around the bug since *self* stores a strong reference.

I didn't see how to write a test which does crash without the fix.

I played with `weakref.ref` to check indirectly if we are called with a strong reference, but it made the test too complicated/fragile for little benefits. I gave up on this way.